virtcontainers/device/drivers/block.go

```diff
@@ -67,14 +67,25 @@ func (device *BlockDevice) Attach(devReceiver api.DeviceReceiver) (err error) {
 		Index:  index,
 	}
 
-	driveName, err := utils.GetVirtDriveName(index)
-	if err != nil {
-		return err
-	}
-
 	customOptions := device.DeviceInfo.DriverOptions
-	if customOptions != nil && customOptions["block-driver"] == "virtio-blk" {
-		drive.VirtPath = filepath.Join("/dev", driveName)
+	if customOptions != nil {
+		if customOptions["block-driver"] == "virtio-blk" {
+			driveName, err := utils.GetVirtDriveName(index)
+			if err != nil {
+				return err
+			}
+			drive.VirtPath = filepath.Join("/dev", driveName)
+		}
+		if customOptions["block-driver"] == "virtio-mmio" {
+			//With firecracker the rootfs for the VM itself
+			//sits at /dev/vda
+			//TODO: Track the VM rootfs if it is a block device
```

Comment:
Could you raise an issue and reference the URL here?

Reply:
Raised and indicated. https://github.com/kata-containers/runtime/issues/1061